### PR TITLE
Sara Dahlström Project Pizza

### PR DESCRIPTION
**Pull request by saradahlstrom**

### CODE REVIEW

**Comment by saradahlstrom** [comment]
Pull request

File: script.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        break;
+      case "2":
+        foodSubGroup = "calzone";
+        foodPrice = 120;
+        break;
+      case "3":
+        foodSubGroup = "quattro stagioni";
+        foodPrice = 150;
+        break;
+    }
```
Comment (Sheryll-Ann) [inline script.js:39]: Something to ponder upon: would it be a good idea to have a "`default case`" at the end of your switches?
Here for example, letters or integers>3 as values for `foodSubGroup` do not prompt any error message, and your code runs to the end with weird values (the random ones entered appear in your final message: `You are about to order M.  The cost will be undefined kronor`).

File: script.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      foodGroup = "pizza";
+      break;
+    case "2":
+      foodGroup = "pasta";
+      break;
+    case "3":
+      foodGroup = "salad";
+      break;
+  }
+
```
Comment (Sheryll-Ann) [inline script.js:20]: When entering another alternative than your 3 conditions, your code runs up to line 84 with the alert message "invalid food choice group" where, at line 83, you close your `switch (foodGroup)` code that starts on line 9.  Was that intended to be this apart?

File: script.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        break;
+      case "2":
+        foodSubGroup = "vongole";
+        foodPrice = 90;
+        break;
+      case "3":
+        foodSubGroup = "ravioli";
+        foodPrice = 100;
+        break;
+    }
```
Comment (Sheryll-Ann) [inline script.js:55]: same suggestion here for same reasons mentioned above: `default` case?

File: script.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        break;
+      case "2":
+        foodSubGroup = "cheese and ham";
+        foodPrice = 80;
+        break;
+      case "3":
+        foodSubGroup = "garden";
+        foodPrice = 70;
+        break;
+    }
```
Comment (Sheryll-Ann) [inline script.js:71]: same suggestion for above reasons about `default` case.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  // Calculate childFoodPrice after foodPrice is assigned a value
+  childFoodPrice = foodPrice / 2;
+
+  var age = prompt("How old is the person that will be eating?");
+  if (age <= 15) {
+    alert(`You are about to order ${foodSubGroup} from our children's menu. The cost will be ${childFoodPrice} kronor. Hope you like it as much as we do and welcome back again!`);
+  } else {
```
Comment (Sheryll-Ann) [inline script.js:80]: Here the user has the possibility to:
- enter a negative integer, a zero for `age`, your result will still be validated by your first condition `(age <= 15)` 
- leave `age` empty, your value displays as `Nan` kronor in your final message
- enter a letter for `age`, your value displays as normal/adult price defined in your `switch (foodSubGroup)`
( I myself omitted the first twos in my code by mistake... )

Suggestion: you can use explicit conversions and narrow down your conditions further e.g.
`var age =Number( prompt("How old is the person that will be eating?"));`
```
if (age >=0 && age <=15) {
    alert(`You are about to order ${foodSubGroup} from our children's menu. The cost will be ${childFoodPrice} kronor. Hope you like it as much as we do and welcome back again!`);
  }

```
and/or add an `else` condition to your last `if...else` statement for any unintended values passed to `age`.

**Review by Sheryll-Ann — COMMENTED** [review]
The code is written in a very straightforward way- short and concise ( I am not but striving to be so :) ).
Errors in the code nevertheless occur because default cases have been omitted.  I think if you read through my code review the different examples I mentioned, you'll understand what I mean.
Good job Sara!  Keep it up!  Was nice reviewing your code!

**Pull request closed without merging**